### PR TITLE
[jit] move cpp tests to python extension

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24458 [jit] move cpp tests to python extension**

Trying to avoid ODR violations in fbcode opt builds. dogsciencing.

Confirmed that this fixes the internal build failures. I also had to remove duplicate inclusion `torch/csrc/api/src/jit.cpp` (it was in both the python extension and libtorch).
This is a remake of https://github.com/pytorch/pytorch/pull/19658

Differential Revision: [D16854303](https://our.internmc.facebook.com/intern/diff/D16854303)

### CODE REVIEW

**Comment by suo** [comment]
I confirmed this works and finally stops our opt/tsan/dbg-cov builds from failing! (they've been failing for like 8 months lol)

File: tools/build_variables.py [hunk 1]
```diff
@@ -117,7 +117,6 @@
     "torch/csrc/jit/register_special_ops.cpp",
     "torch/csrc/jit/scope.cpp",
     "torch/csrc/jit/script/compiler.cpp",
-    "torch/csrc/api/src/jit.cpp",
```
Comment (ZolotukhinM) [inline tools/build_variables.py]: This one disappeared, is it intentional?

**Review by ZolotukhinM — COMMENTED** [review]
It'd be nice to understand what was failing. I also wonder if it was fixed by the removing `torch/csrc/api/src/jit.cpp` or by the move of `test/cpp/jit/test.cpp` to python extensions :)

File: tools/build_variables.py (hunk 1 above)
Comment (suo) [reply tools/build_variables.py]: This was a second problem uncovered by fixing the first; this file was added to both the python extension and libtorch sources, so we got ODR violations.

**Review by ZolotukhinM — APPROVED** [review]
I see, looks good then (might be a good idea to add your explanation to the commit message 😛).

**Comment by suo** [comment]
closing in favor of https://github.com/pytorch/pytorch/pull/24801, which is a more fundamental fix

**Pull request closed without merging**